docs/stream-time-from-player-time.md

```diff
@@ -0,0 +1,128 @@
+* GOP Fusion
+  * Used when: the segment doesn't start with a keyframe.
+  * Impact: if GOPs were saved from previous segment appends, the last GOP will be prepended to the segment.
+  * Need to account in time conversion? Yes. The segment is artificially extended, so while it shouldn't impact the stream time itself (since it will overlap with content already appended), it will impact the post transmux start of segment.
+* GOPS to Align With
+  * Used when: switching renditions, or appending segments with overlapping GOPs (intersecting time ranges).
+  * Impact: GOPs in the segment will be dropped until there are no overlapping GOPs with previous segments.
+  * Need to account in time conversion? No. So long as we aren't switching renditions, and the content is sane enough to not contain overlapping GOPs, this should not have a meaningful impact.
+
+Among the changes, with only GOP Fusion having an impact, the task is simplified. Instead of accounting for any changes to the video stream, only those from GOP Fusion should be accounted for. Since GOP fusion will potentially only prepend frames to the segment, we just need the number of seconds prepended to the segment when offsetting the time. As such, we can add the following properties to each segment:
+
+```
+segment: {
+  // calculated start of segment from either end of previous segment or end of last buffer
+  // (in stream time)
+  start,
+  ...
+  videoTimingInfo: {
+    // number of seconds prepended by GOP fusion
+    transmuxerPrependedSeconds
+    // start of transmuxed segment (in player time)
+    transmuxedPresentationStart
+  }
+}
+```
+
+## The Formula
+
+With the properties listed above, calculating a *stream time* from a *player time* is given as follows:
+
+```
+const playerTimeToStreamTime = (playerTime, segment) => {
+  // If there's no "anchor point" for the stream time (i.e., a time that can be used to
+  // sync the start of a segment with a real world stream time), then a stream time can't
+  // be calculated.
+  if (!segment.dateTimeObject) {
+    return null;
+  }
+
+  const transmuxerPrependedSeconds = segment.videoTimingInfo.transmuxerPrependedSeconds;
+  const transmuxedStart = segment.videoTimingInfo.transmuxedPresentationStart;
+
+  // get the proper start of new content (not prepended old content) from the segment,
+  // in player time
+  const startOfSegment = transmuxedStart + transmuxerPrependedSeconds;
+  const offsetFromSegmentStart = playerTime - startOfSegment;
+
+  return new Date(segment.dateTimeObject.getTime() + offsetFromSegmentStart * 1000);
+};
+```
+
+The *stream time* can be converted to *program time* by taking the EXT-X-PROGRAM-DATE-TIME tagged on the segment and adding *stream time* - segment.start.
+
+## Examples
+
+```
+// Stream Times:
+//   segment1: 30.1 => 32.1
+//   segment2: 32.1 => 34.1
+//   segment3: 34.1 => 36.1
+//
+// Player Times:
+//   segment1: 0 => 2
+//   segment2: 2 => 4
+//   segment3: 4 => 6
+
+const segment2 = {
```

Comment:
For now `transmuxerPrependedSeconds` is guaranteed to be 0 for the first segment, though it could change in the future (if, for instance, we get into prepending blank video frames to start a video with offset timing).